state-chain/primitives/src/chains/assets.rs

```diff
@@ -345,6 +345,12 @@ macro_rules! assets {
 						iter.clone().find(|(asset, _t)| *asset == required_asset).ok_or(()).map(|x| x.1)
 					}).ok()
 				}
+
+				pub fn iter(&self) -> impl Iterator<Item = (Asset, &T)> {
+					Asset::all().map(|asset| {
+						(asset, &self[asset])
```

Comment:
There's no static guarantee that self[asset] exists (is there?) - it's true as long as the macros are correct, but maybe it's worth including a unit test for each asset map that just does an `iter().collect()` to prevent mistakes being introduced.

Reply:
it is guaranteed by the function signature the lack of panics in the index impl. But the code could have more unit tests so ill add some more.